src/player.cpp

```diff
@@ -8484,6 +8484,7 @@ void player::vomit()
         // Remove all joy form previously eaten food and apply the penalty
         rem_morale( MORALE_FOOD_GOOD );
         rem_morale( MORALE_FOOD_HOT );
+        rem_morale( MORALE_FOOD_CARNIVORE );
```

Comment:
rem_morale( MORALE_FOOD_COLD ); is missing from here. No idea if that was intentional. It does something similar to rem_morale( MORALE_FOOD_HOT ); Should I add it?